From deedce75fcfdfa9f81c6be5d75f878df2d28adac Mon Sep 17 00:00:00 2001
From: Colin Cross <ccross@android.com>
Date: Wed, 9 May 2012 16:09:50 -0700
Subject: [PATCH] trace: power: add trace_clock_set_parent

Adds a new trace event to be called from clk_set_parent.  Some
cpufreq drivers, including Tegra, reparent the cpu clock to a
slower clock while the main pll is relocking, tracing
clk_set_parent allows traces to show how for long the cpu is
running slower.

Uses a separate TRACE_EVENT instead of the clock event class to
allow the event to contain string names for the child and the
parent.

Signed-off-by: Colin Cross <ccross@android.com>
---
 include/trace/events/power.h | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/include/trace/events/power.h b/include/trace/events/power.h
index e5bf9a7..f2ec464 100644
--- a/include/trace/events/power.h
+++ b/include/trace/events/power.h
@@ -232,6 +232,25 @@ DEFINE_EVENT(clock, clock_set_rate,
 	TP_ARGS(name, state, cpu_id)
 );
 
+TRACE_EVENT(clock_set_parent,
+
+	TP_PROTO(const char *name, const char *parent_name),
+
+	TP_ARGS(name, parent_name),
+
+	TP_STRUCT__entry(
+		__string(       name,           name            )
+		__string(       parent_name,    parent_name     )
+	),
+
+	TP_fast_assign(
+		__assign_str(name, name);
+		__assign_str(parent_name, parent_name);
+	),
+
+	TP_printk("%s parent=%s", __get_str(name), __get_str(parent_name))
+);
+
 /*
  * The power domain events are used for power domains transitions
  */
-- 
1.8.0

